### ISSUE TITLE
Missing Turn Lane Indication at specific Way

### ISSUE THREAD

**Issue by Peter0** [osmandapp/OsmAnd#21182]
### Routing engine

- [X] OsmAnd's in-app offline routing
- [ ] Any online routing provider (YOURS, openrouteservice, OSRM, etc.)

### Routing Profile

Car

### Start and end points

Start: 50.72367 7.11313
End: 50.71716 7.11985

### Actual route

The route itself goes straight on; this is ok.
When I drive past Way 1078462627, Osmand only shows me the Turn Lane Indication (screen overlay with yellow & grey arrows) if the planned route leads to the right-turn lane. Using the above given destination, the route leads straight on, and there is no turn lane indication here – other than usually, when there are several directional lanes - even if it is supposed to go straight ahead.

### Expected route

I expect, when about to pass the mentioned Way, two straight yellow arrows, adjacent to one grey one, heading slightly right.

### Maps you used (online or offline)

- [×] Offline maps offered within the OsmAnd app for download. 
- [ ] Online (tile / raster) maps 
Standard map Europe/Germany/NRW/Regierungsbezirk Köln.
Live Updates are active.
destination tags are only in live update data, but turn lane tags are quite old, they should be among monthly data.
Even when removing live updates, there are still no arrows.

### Your Environment

```
OsmAnd Version: ~4.6.8
Android version: 9
Device model: Samsung Galaxy J5

```

**Comment by yuriiurshuliak** [comment]
The bug could not be reproduced. I reviewed the route you provided using our [web version](https://test.osmand.net/map/navigate/?start=50.723670,7.113130&end=50.717160,7.119850&profile=car#15/50.7220/7.1136) and didn’t observe any missing turn indicators.

Could you please specify the location where you encounter incorrect or missing turn instructions? Additionally, attach a screenshot highlighting the issue.

**Comment by Peter0** [comment]
This is when I chose a dest. with turn.
![Screenshot_Turn](https://github.com/user-attachments/assets/128ded7e-7230-4935-9ad2-4d2486daf2dc)
Note the circled turn lane hint "in 150 m"

![Screenshot_Straight](https://github.com/user-attachments/assets/9c990f6a-b863-405b-b88b-6f95af609e28)

Here is the issue: I expect to have a similar turn hint in 150 m, just coloured the other way (straight arrows yellow).

Just to be sure, I made a new profile of type car with default settings. There I noted no differences.

**Comment by vshcherb** [comment]
please check https://test.osmand.net/map/navigate/?start=50.723606,7.112886&end=50.719626,7.114055&profile=car#17/50.72019/7.11659 - looks correct to me

**Comment by DmitryAlexei** [comment]
@Peter0 It seems that your issue is similar to https://github.com/osmandapp/OsmAnd/issues/19065
Hopefully, new settings "Display on minor intersections when navigating straight" will be added to "Turn lanes widget" in 4.9 release. And so your issue will be resolved.

**Comment by Peter0** [comment]
> @Peter0 It seems that your issue is similar to #19065

Haven't read the thread thoroughly, but the initial described behaviour looks similar. Especially, it pointed my awareness to the oneway feature.
Obviously I have no problem with the widget not popping up at all (contrary to some of the participants there) – even on the "bad" screenshot it is visible, albeit with the 500 m hint instead of the 150 m one.

My reference case is passing https://www.openstreetmap.org/way/1116387389 in forward direction. There is usually no voice, but the widget assures one that straight, not left, is the direction of choice.

**Comment by Peter0** [comment]
> please check https://test.osmand.net/map/navigate/?start=50.723606,7.112886&end=50.719626,7.114055&profile=car#17/50.72019/7.11659 - looks correct to me

I don’t know the meaning of the symbols, but I suppose that the grey and orange circles represent actions on the screen. If I go to https://test.osmand.net/map/navigate/?start=50.723606,7.112886&end=50.715972,7.121308&profile=car#17/50.71926/7.11543 I miss a circle near housenumber 170.

**Comment by vshcherb** [comment]
@yuriiurshuliak please upload simulated route using latest version

**Comment by yuriiurshuliak** [comment]
The bug has been confirmed. To reproduce, build a route using these coordinates: [Navigation Route](https://test.osmand.net/map/navigate/?start=50.723606,7.112886&end=50.719626,7.114055&profile=car#15/50.7201/7.1158) and run a navigation simulation with both a right turn and a straight path.

The result shows that for the right turn, both Next turn and Lanes display incorrect turn information. Similarly, when moving straight, the Lanes widget incorrectly shows a turn at the same location.

OsmAnd~ 5.0.0#3656m, released: 2024-11-05

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/2a1e996d-7432-4497-a1ea-48f7fd2ffbf1"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/bf351497-4124-4a10-8ad2-07d8e4840ebc"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by Peter0** [comment]
No. In right-turn-case both turn indication as well as lanes display look sensible to me.

**Comment by Peter0** [comment]
And also the lanes display at the straight case look perfectly sensible.
My only wish is that the "use one of the two straight lanes on the left, don't take the slight right turn lane on the right side" lanes display shall be just passed (at the very moment shown on the right screenshot from posting 14 h ago).

**Comment by vshcherb** [comment]
Lane indication will be added as setting with https://github.com/osmandapp/OsmAnd/issues/19065 . SLight right turn is for another road https://www.openstreetmap.org/way/178473299.
